packages/element-web-module-api/src/api/custom-components.ts

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import type { JSX } from "react";
+import type { MatrixEvent } from "matrix-js-sdk";
+
+/**
+ * Targets in Element for custom components.
+ * @public
+ */
+export enum CustomComponentTarget {
+    /**
+     * Component that renders "m.room.message" events in the room timeline.
+     */
+    TextualBody = "TextualBody",
+}
```

Comment:
Wouldn't a pattern which lets you specify the event type you wish to register a custom component renderer for make more sense and be more reusable and generic? Like what if a module wanted to add a renderer to `org.company.custom-data` events?

Reply:
I don't think that's always the case thought. You might find you want to override all message types (e.g. you've got a requirement to show an eventId below the message, or other metadata), or in my specific demo case I actually care about the contents *of* the message and not the type. 

In the hookshot example it uses m.room.messages, but it might switch in the future. What we want to render is an info box *if* the message contains a 

```
    "org.matrix.matrix-hookshot.openproject.project": {
      "id": 1,
      "name": "Demo project",
      "url": "https://elementio-demo.openproject.com/projects/1"
    },
```

then we want to render those details.

Reply:
Then supporting wildcards sounds sane, where the OriginalComponent would vary depending on if that event has a nominal renderer or not